src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,720 @@
+        for kv in fetched_items:
+            key = kv.key
+            if prefix is not None:
+                if not key.startswith(prefix):
+                    continue
+                key = key[len(prefix):]
+
+            if previous_kv is not None and previous_kv.key == key:
+                if previous_kv.value != kv.value:
+                    raise CLIError(
+                        "The key {} has two labels {} and {}, which conflicts with each other.".format(previous_kv.key, previous_kv.label, kv.label))
+                continue
+            previous_kv = KeyValue(key, kv.value)
+
+            # No need to construct for properties format
+            if format_ == 'properties' or separator is None or not separator:
+                exported_dict.update({key: kv.value})
+                continue
+
+            key_segments = key.split(separator)
+            if key_segments[0].isdigit():
+                __export_keyvalue(key_segments,
+                                  kv.value, exported_list, key)
+
+            else:
+                __export_keyvalue(key_segments,
+                                  kv.value, exported_dict, key)
+
+        if exported_dict and exported_list:
+            logger.error("Can not export to a valid file! Some keys have been dropped. %s", json.dumps(
+                exported_dict, indent=2, ensure_ascii=False))
+
+        return __compact_key_values(exported_dict if not exported_list else exported_list)
+    except Exception as exception:
+        raise CLIError("Fail to export key-values." + str(exception))
+
+
+def __flatten_dict_to_properties(value_to_flatten, separator='.', prefix=''):
+    flattened_properties_dict = {}
+    if isinstance(value_to_flatten, dict):
+        for keys, values in value_to_flatten.items():
+            current_prefix = prefix + separator + str(keys) if prefix else str(keys)
+            if isinstance(values, (dict, list)):
+                flattened_properties_dict.update(__flatten_dict_to_properties(values, separator, current_prefix))
+            else:
+                flattened_properties_dict.update({current_prefix: str(values)})
+
+    elif isinstance(value_to_flatten, list):
+        for idx, val in enumerate(value_to_flatten):
+            current_prefix = prefix + '[' + str(idx) + ']' if prefix else str(idx)
+            if isinstance(val, (dict, list)):
+                flattened_properties_dict.update(__flatten_dict_to_properties(val, separator, current_prefix))
+            else:
+                flattened_properties_dict.update({current_prefix: str(val)})
+    else:
+        flattened_properties_dict.update({prefix: str(value_to_flatten)})
+
+    return flattened_properties_dict
+
+
+def __export_features(retrieved_features, format_):
+    exported_dict = {}
+    client_filters = []
+    if format_ == 'json':
+        exported_dict["FeatureManagement"] = {}
+
+    elif format_ in ('yaml', 'properties'):
+        # Currently, this condition will never be met
```

Comment:
add log warning for formats not supported

Reply:
sounds good.

---
In reply to: [356322906](https://github.com/Azure/azure-cli/pull/11467#discussion_r356322906) [](ancestors = 356322906)